File: vane-portals/src/main/java/org/oddlama/vane/portals/event/PortalSelectTargetEvent.java
package org.oddlama.vane.portals.event;

import org.bukkit.entity.Player;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.Nullable;
import org.oddlama.vane.portals.portal.Portal;

public class PortalSelectTargetEvent extends PortalEvent {

    private static final HandlerList handlers = new HandlerList();
    private Player player;
    private Portal portal;
    private Portal target;
    private boolean checkOnly;

    public PortalSelectTargetEvent(final Player player, final Portal portal, final Portal target, boolean checkOnly) {
        this.player = player;
        this.portal = portal;
        this.target = target;
        this.checkOnly = checkOnly;
    }

    public Player getPlayer() {
        return player;
    }

    public Portal getPortal() {
        return portal;
    }

    public @Nullable Portal getTarget() {
        return target;
    }

    public boolean checkOnly() {
        return checkOnly;
    }

    public HandlerList getHandlers() {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: vane-portals/src/main/kotlin/org/oddlama/vane/portals/event/PortalSelectTargetEvent.kt
package org.oddlama.vane.portals.event

import org.bukkit.entity.Player
import org.bukkit.event.HandlerList
import org.oddlama.vane.portals.portal.Portal

/**
 * Fired before a portal target is changed or selected.
 *
 * @property player the acting player.
 * @property portal the source portal.
 * @property target the requested target portal.
 * @property checkOnly whether this is a dry-run validation.
 */
class PortalSelectTargetEvent(
    @JvmField val player: Player?,
    @JvmField val portal: Portal?,
    @JvmField val target: Portal?,
    private val checkOnly: Boolean
) : PortalEvent() {
    /** Returns whether this event is in validation-only mode. */
    fun checkOnly() = checkOnly

    /** Returns Bukkit handlers for this event. */
    override fun getHandlers() = handlerList

    /** Static Bukkit handler list for this event type. */
    companion object {
        /** Shared handler list used by Bukkit's event system. */
        @JvmStatic
        val handlerList: HandlerList = HandlerList()
    }
}
